tests/test_model.py

```diff
@@ -454,18 +454,39 @@ def finalize(self, status):
 
         assert simple_logger_more_saves.saved_count > simple_logger_less_saves.saved_count
 
-    @mock.patch.object(MatrixFactorizationModel, 'calculate_loss')
     def test_terminate_on_nan(self,
-                              calculate_loss_mock,
                               train_val_implicit_sample_data):
-        calculate_loss_mock.return_value = torch.tensor(float('nan')).requires_grad_()
-
         train, val = train_val_implicit_sample_data
         model = MatrixFactorizationModel(train=train, val=val)
-        trainer = CollieMinimalTrainer(model=model, terminate_on_nan=True)
 
-        with pytest.raises(ValueError):
-            trainer.fit(model)
+        with pytest.warns(
+            DeprecationWarning,
+            match=r"``terminate_on_nan`` is deprecated and was replaced with ``detect_anomaly``."
+        ):
+            trainer = CollieMinimalTrainer(model=model, terminate_on_nan=True)
+
+        assert trainer.detect_anomaly
+
+    def test_detect_anomaly(self,
+                            train_val_implicit_sample_data):
+        def bad_loss(*args, **kwargs):
+            return torch.tensor(0.0, requires_grad=True) * torch.tensor(float('nan'))
```

Comment:
oof, another nasty debugging session. Might be worth leaving a comment here on why you went with this approach.